Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionWindow/MotionPropertiesWindow.cpp

```diff
@@ -7,51 +7,42 @@
  */
 
 #include <AzCore/StringFunc/StringFunc.h>
-#include <Editor/InspectorBus.h>
 #include <EMotionFX/CommandSystem/Source/CommandManager.h>
 #include <EMotionStudio/EMStudioSDK/Source/EMStudioManager.h>
 #include <EMotionStudio/Plugins/StandardPlugins/Source/MotionWindow/MotionExtractionWindow.h>
-#include <EMotionStudio/Plugins/StandardPlugins/Source/MotionWindow/MotionListWindow.h>
-#include <EMotionStudio/Plugins/StandardPlugins/Source/MotionWindow/MotionRetargetingWindow.h>
 #include <EMotionStudio/Plugins/StandardPlugins/Source/MotionWindow/MotionPropertiesWindow.h>
-#include <EMotionStudio/Plugins/StandardPlugins/Source/MotionWindow/MotionWindowPlugin.h>
-#include <EMotionStudio/Plugins/StandardPlugins/Source/MotionSetsWindow/MotionSetsWindowPlugin.h>
+#include <Editor/InspectorBus.h>
 #include <QPushButton>
 #include <QVBoxLayout>
 
 namespace EMStudio
 {
-    MotionPropertiesWindow::MotionPropertiesWindow(QWidget* parent, MotionWindowPlugin* motionWindowPlugin)
+    MotionPropertiesWindow::MotionPropertiesWindow(QWidget* parent)
         : QWidget(parent)
-        , m_motionWindowPlugin(motionWindowPlugin)
     {
-        // motion properties
         QVBoxLayout* motionPropertiesLayout = new QVBoxLayout(this);
         motionPropertiesLayout->setMargin(0);
         motionPropertiesLayout->setSpacing(0);
 
-        // add the motion extraction stack window
-        m_motionExtractionWindow = new MotionExtractionWindow(this, m_motionWindowPlugin);
+        m_motionExtractionWindow = new MotionExtractionWindow(this);
         m_motionExtractionWindow->Init();
         AddSubProperties(m_motionExtractionWindow);
 
-        // add the motion retargeting stack window
-        m_motionRetargetingWindow = new MotionRetargetingWindow(this, m_motionWindowPlugin);
-        m_motionRetargetingWindow->Init();
-        AddSubProperties(m_motionRetargetingWindow);
-
         FinalizeSubProperties();
+
+        GetCommandManager()->RegisterCommandCallback<CommandSelectCallback>("Select", m_callbacks, this, false);
     }
 
     MotionPropertiesWindow::~MotionPropertiesWindow()
     {
         // Clear the inspector in case this window is currently shown.
         EMStudio::InspectorRequestBus::Broadcast(&EMStudio::InspectorRequestBus::Events::ClearIfShown, this);
-    }
 
-    void MotionPropertiesWindow::UpdateMotions()
-    {
-        m_motionRetargetingWindow->UpdateMotions();
+        for (auto callback : m_callbacks)
```

Comment:
question: Is this making a copy on each iteration of the loop? If it's a pointer could we use `auto*` to make it a little clearer to the reader. Thanks!

Reply:
Whups, good call. This indeed is a pointer - changed to auto*.